x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/history/HistoryActionConditionTests.java

```diff
@@ -115,9 +115,14 @@ public void testActionConditionWithHardFailures() throws Exception {
         assertWatchWithMinimumActionsCount(id, ExecutionState.EXECUTED, 1);
 
         // only one action should have failed via condition
-        final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
-        assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
+        assertBusy(() -> {
+            // Watcher history is now written asynchronously, so we check this in an assertBusy
+            ensureGreen(HistoryStoreField.DATA_STREAM);
```

Comment:
Is the `ensureGreen` inside the `assertBusy` because the index may not exist (and would throw an assertion failure?)

Reply:
Yes. I haven't actually seen it on this test, but the test in https://github.com/elastic/elasticsearch/pull/95642 failed because it tried to query the index before the shards were allocated. I figured no harm in checking that here -- it might save us from a rare test failure.

Reply:
And I think if we try to query the index before shards are allocated it will throw an uncaught exception rather than an assertion failure, so the assertBusy block would fail.